src/main/java/de/refactoringbot/refactoring/RefactoringHelper.java

```diff
@@ -4,333 +4,289 @@
 	 */
-	public static String getFullMethodSignature(MethodDeclaration methodDeclaration) {
-		ResolvedMethodDeclaration resolvedMethod = methodDeclaration.resolve();
-		return resolvedMethod.getQualifiedSignature();
+	public static boolean isFieldDeclarationAtLine(FieldDeclaration fieldDeclaration, Integer lineNumber) {
+		Optional<Position> beginPositionOfField = fieldDeclaration.getBegin();
+		return (beginPositionOfField.isPresent() && beginPositionOfField.get().line == lineNumber);
 	}
 
 	/**
 	 * @param methodDeclaration
-	 * @param position
-	 * @return true if given method starts at given position, false otherwise
+	 * @return parent node of the given method declaration as
+	 *         ClassOrInterfaceDeclaration
+	 * @throws IllegalStateException
+	 *             if no parent node is present
 	 */
-	public static boolean isMethodDeclarationAtLine(MethodDeclaration methodDeclaration, Integer position) {
-		if (methodDeclaration.getName().getBegin().isPresent()) {
-			if (position == methodDeclaration.getName().getBegin().get().line) {
-				return true;
-			}
+	public static ClassOrInterfaceDeclaration getMethodParentNodeAsClassOrInterface(
+			MethodDeclaration methodDeclaration) {
+		Optional<Node> parentNode = methodDeclaration.getParentNode();
+		if (parentNode.isPresent()) {
+			return ((ClassOrInterfaceDeclaration) parentNode.get());
 		}
-		return false;
+		throw new IllegalStateException("MethodDeclaration expected to have a parent node.");
 	}
 
 	/**
-	 * @param methodDeclaration
-	 * @return the local signature of a method as a string
+	 * @param filePath
+	 * @return all <code>ClassOrInterfaceDeclaration</code> in the given file
+	 * @throws FileNotFoundException
 	 */
-	public static String getMethodSignatureAsString(MethodDeclaration methodDeclaration) {
-		return methodDeclaration.getSignature().asString();
+	public static List<ClassOrInterfaceDeclaration> getAllClassesAndInterfacesFromFile(String filePath)
+			throws FileNotFoundException {
+		FileInputStream is = new FileInputStream(filePath);
+		CompilationUnit cu = LexicalPreservingPrinter.setup(JavaParser.parse(is));
+		return cu.findAll(ClassOrInterfaceDeclaration.class);
 	}
 
 	/**
-	 * This method gets all direct and indirect Ancestors of a given class if
-	 * possible. (If ancestor is not a external dependency for example)
-	 * 
-	 * @param currentClass
-	 * @return ancestors
+	 * @param allJavaFiles
+	 * @param targetClass
+	 * @return list of qualified class or interface names which are reachable via
+	 *         the inheritance hierarchy of the given class (ancestors, descendants,
+	 *         siblings, ...)
+	 * @throws BotRefactoringException
+	 * @throws FileNotFoundException
 	 */
-	public static List<ResolvedReferenceType> getAllAncestors(ResolvedReferenceTypeDeclaration currentClass) {
-		// Init ancestor list
+	public static Set<String> findQualifiedNamesOfRelatedClassesAndInterfaces(List<String> allJavaFiles,
```

Comment:
Do we need this method? It just gets all ancestors and calls a method with the same that also uses those ancestors. Can't we just call the second method directly and find all the ancestors there?

Reply:
Yes, good point.